src/pages/nft/claim/index.vue

```diff
@@ -1223,14 +1231,49 @@ export default {
         );
       }
     },
-    handleViewCollection() {
+    async ensureClassIdInCollected() {
+      this.isViewCollectionLoading = true;
+
+      let collectedNFTs = this.getCollectedNFTClassesByAddress(
+        this.claimingAddress
+      );
+
+      for (let attempts = 0; attempts < MAX_ATTEMPT; attempts += 1) {
+        try {
+          // eslint-disable-next-line no-await-in-loop
+          await this.fetchCollectedNFTClassesByAddress({
+            address: this.claimingAddress,
+            nocache: true,
+          });
+
+          collectedNFTs = this.getCollectedNFTClassesByAddress(
+            this.claimingAddress
+          );
+
+          if (collectedNFTs?.some(item => item.classId === this.classId)) {
+            break;
+          }
+          // eslint-disable-next-line no-await-in-loop
+          await sleep(500);
```

Comment:
The fetch above with `nocache: true` set a qs according to current utm second, so better sleep >= 1000ms

Reply:
[🎨 Increase sleep duration](https://github.com/likecoin/liker-land/pull/1773/commits/e49acdb1a431c7fe1b8337f7c7e4730662d7cfdf)